test/e2e/operator.go

```diff
@@ -321,3 +325,98 @@ var _ = XDescribe("ARO Operator - MachineSet Controller", func() {
 		}
 	})
 })
+
+var _ = Describe("ARO Operator - Azure Subnet Reconciler", func() {
+	var vnetName, location, resourceGroup string
+	var subnetsToReconcile map[string]*string
```

Comment:
I think we do not need a pointer to a string as a value. Just `string` should be ok.

```suggestion
	var subnetsToReconcile map[string]string
```